cli/js/40_jupyter.js

```diff
@@ -412,6 +419,42 @@ function enableJupyter() {
     return;
   }
 
+  /**
+   * Prompt for user confirmation (in Jupyter Notebook context)
+   * Override confirm and prompt because they depend on a tty
+   * and in the Deno.jupyter environment that doesn't exist.
+   * @param {string} message - The message to display.
+   * @returns {Promise<boolean>} User confirmation.
+   */
+  async function confirm(message = "Confirm") {
+    const answer = await input(`${message} [y/N] `, false);
+    return answer === "Y" || answer === "y";
+  }
```

Comment:
This PR looks great, but prompt and confirm are not async functions. Probably we should make `op_jupyter_input` sync, then block on the future in rust.

Reply:
I believe the way to do this is to do the following inside the op:

```rs
tokio::runtime::Handle::current().block_on(async {
 // async code goes here
})
```

Reply:
I tried convering to `block_on` and hit the following error:

```
thread 'main' panicked at cli/ops/jupyter.rs:53:46:
Cannot start a runtime from within a runtime. This happens because a function (like `block_on`) attempted to block the current thread while the thread is being used to drive asynchronous tasks.
```

I'll dig into that further and if you have advice from already having more context let me know!

Reply:
So apparently the answer is this isn't possible to do with a single threaded runtime. We'll have to move a bunch of the messaging off into a separate thread then make this other code have the ability to be sync.

Reply:
I opened https://github.com/denoland/deno/issues/23617

Reply:
Ok! 👍 . Thanks for opening up the other feature request.